src/core/renderer.js

```diff
@@ -160,7 +160,9 @@ class Renderer {
         const strokeColorRequirements = style.getStrokeColor()._getDrawMetadataRequirements();
         const strokeWidthRequirements = style.getStrokeWidth()._getDrawMetadataRequirements();
         const filterRequirements = style.getFilter()._getDrawMetadataRequirements();
-        let requiredColumns = [widthRequirements, colorRequirements, strokeColorRequirements, strokeWidthRequirements, filterRequirements]
+        const styleVariables = style._styleSpec.variables;
```

Comment:
If you create the `getVariables` function you won't be accessing a private field here.